internal/pki/pki_test.go

```diff
@@ -15,32 +13,63 @@ package pki
  limitations under the License.
 */
 
+package pki
+
 import (
+	"crypto/ecdsa"
 	"crypto/x509"
-	"net"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
 	"testing"
+	"time"
 
 	"gotest.tools/v3/assert"
+
+	"github.com/crunchydata/postgres-operator/internal/testing/require"
 )
 
+type StringSet map[string]struct{}
+
+func (s StringSet) Has(item string) bool { _, ok := s[item]; return ok }
+func (s StringSet) Insert(item string)   { s[item] = struct{}{} }
+
+func TestCertificateCommonName(t *testing.T) {
+	zero := Certificate{}
+	assert.Assert(t, zero.CommonName() == "")
+}
+
+func TestCertificateDNSNames(t *testing.T) {
+	zero := Certificate{}
+	assert.Assert(t, zero.DNSNames() == nil)
+}
+
+func TestCertificateHasSubject(t *testing.T) {
+	zero := Certificate{}
+
+	// The zero value has no subject.
+	for _, cn := range []string{"", "any"} {
+		for _, dns := range [][]string{nil, {}, {"any"}} {
+			assert.Assert(t, !zero.hasSubject(cn, dns), "for (%q, %q)", cn, dns)
```

Comment:
This feels like it is written to test a variety of cn and dns, but won't `hasSubject` set ok == false early?

Reply:
Yes, it does. Any arguments have the same outcome.

Reply:
When I remove `c.x509 != nil` from the lines you highlighted, this test panics and other tests pass. So this test is the only time `hasSubject` is called with a zero value.

🤔 That means the other exported functions (probably) don't hit this path, but it feels irresponsible to use the pointer without first checking if it is nil.

I think both the check and the test should stay. You're right that the one line is covered regardless of the arguments passed in, but this seems like a quick enough way to show that in the test. Should there be fewer assertions? Drop the loops and do just a couple?

```go
assert.Assert(t, !zero.hasSubject("", nil))
assert.Assert(t, !zero.hasSubject("any", []string{"thing"}))
```